ingest-app/src/cmr/ingest/services/bulk_update_service.clj

```diff
@@ -106,26 +107,46 @@
                                             update-type)]))))
 
 (defn- get-provider-concept-ids
-  "Returns a list of collection concept ids for a given provider-id"
+  "Returns a list of collection concept ids for a given provider-id.
+   Throws exception if non are returned."
   [context provider-id]
   (let [concepts (mdb/find-concepts context 
                                     {:provider-id provider-id} 
                                     :collection)]
-    (when (seq concepts)
-      (distinct (map :concept-id concepts)))))
+    (if (seq concepts)
+      (distinct (map :concept-id concepts))
+      (errors/throw-service-errors
+        :bad-request
+        ["No concept-ids are associated with the provider-id."]))))
+
+(defn- validate-concept-ids
+  "Throws exception if there exists invalid
+   concept-ids."
+  [concept-ids]
+  (when-not (= ["ALL"] (map string/upper-case concept-ids))
+    (let [err-msgs 
+           (for [id concept-ids
+                 :let [err-msg (concepts/concept-id-validation id)
+                       err-msg (if err-msg
+                                 err-msg
+                                 (when-not (string/starts-with? id "C")
+                                   [(str "Concept-id [" id 
```

Comment:
use `format` here would be better than `str`.

Reply:
Used str because the line is getting too long...   Will use a separate function then *Sign*